src/posts/2020-08-20-an-indepth-guide-to-aria-roles.md

```diff
@@ -0,0 +1,190 @@
+
+Document roles are used to provide a structural description for a section of content. A majority of the available roles in this category resemble available semantic HTML tags and it is recommended to only use them when an appropriate native tag is not available. The roles in this category are also not typically interactive in nature, but may be used to contain other roles that are interactive. The `toolbar"` and `tooltip"` roles, demonstrated below, serve as good examples of when to use Document Structure roles to provide information to assistive technologies when the equivalent native HTML tag is not available.
+
+#### Toolbar example
+
+```html
+<div role="toolbar">
+  <div class="text-characteristics">
+    <button>Bold</button>
+    <button>Italic</button>
+    <button>Underline</button>
+  </div>
+
+  <div class="text-alignment">
+    <button>Left</button>
+    <button>Middle</button>
+    <button>Right</button>
+  </div>
+</div>
+```
+
+#### Tooltip example
+
+```html
+<button>Show Tooltip</button>  
+<div role="tooltip">Notifications</div>  
+```
+
+- View the [full list of available Document Structure roles](https://www.w3.org/TR/wai-aria/#document_structure_roles).
+
+### Landmark Roles
+
+Landmark roles identify content areas within a page, and as the name implies serve as navigational landmarks for the types of content areas that users may be interested in. These content areas include the page's primary content, collections of navigation links, and search functions to name a few.
+
+Most Landmark roles may also be reasonably associated with native HTML tags. For example:
+
+- The `banner` role can be associated with the `<header>` tag
+- The `complementary` role can be associated with the `<aside>` tag
+- The `contentinfo` role can be associated with the `<footer>` tag
+
+Because the support for some of the above native HTML elements may not be supported in older browsers or some screen readers, it is recommended that the role and native element be paired whenever possible.
+
+#### Navigation role example
+
+```html
+<nav role="navigation">
+  <ul>
+    <li><a href="#a">Dexter</a></li>
+    <li><a href="#b">Doctor Who</a></li>
+    <li><a href="#c">Futurama</a></li>
+  </ul>  
+</nav>
+```
+
+#### Form role example
+
+```html
+<form role="search">
+  <input type="text" name="search" />
+  <input type="submit" name="button" value="Search" />
+</form>
+```
+
+#### Footer role example
+
+```html
+<footer role="contentinfo">
+  <p>&copy; 2020 Small Business Ltd. All rights reserved.</p>
+</footer>
+```
+
+- View the [full list of available Landmark roles](https://www.w3.org/TR/wai-aria/#landmark_roles).
+
+### Live Region Roles
+
+Live Region roles are used to alert screen readers and assistive technologies of dynamic changes to the contents of a page so that they can be announced to users. The reason for this is that dynamic changes may be visually noticeable, but that may not be the case for screen reader users for .
```

Comment:
```suggestion
Live Region roles are used to inform assistive technologies, such as screen readers, of dynamic changes to page content, so that they can be announced to users. The reason for this is that dynamic changes may be visually noticeable, but likely will be missed by users with low or no vision.
```